hack/util.sh

```diff
@@ -460,17 +458,18 @@ function kill_all_processes()
 {
 	local sudo="${USE_SUDO:+sudo}"
 
-	pids=($(jobs -pr))
-	for i in ${pids[@]-}; do
-		pgrep -P "${i}" | xargs $sudo kill &> /dev/null
-		$sudo kill ${i} &> /dev/null
+	for parent in $(jobs -pr); do
+		for child in $(pgrep -P "${parent}"); do
+			"${sudo}" kill "${child}" &> /dev/null
```

Comment:
Sure, you could just append all the pids to a list and `kill -TERM` them all together, sleep, then repeat with `kill -KILL`.